Add support for tags parameter to cloudformation module

psvo opened the pull request:
Expose boto.cloudformation.create_stack() tags parameter. Supplied tags
will be applied to stack and all it's resources on stack creation.
Cannot be updated later (not supported by UpdateStack CloudFormation
API).

psvo committed:
Add support for tags parameter to cloudformation module

Expose boto.cloudformation.create_stack() tags parameter. Supplied tags
will be applied to stack and all it's resources on stack creation.
Cannot be updated later (not supported by UpdateStack CloudFormation
API).

psvo commented:
Ehm, I checked it against bad version of boto. Tags parameter is supported since boto 2.6.0. Would you be interested in modifying patch to check for boto version or there is no chance of merging it?

jimi-c commented:
Yes, if a new feature requires a specific version to work there should be a version check. I believe the ec2 module does that for one param already.

mpdehaan commented:
Thanks for this.

One concern -- this says this takes a dictionary of tags, so it seems to really require an example showing how to pass those in I'd think or people might not know how to use it.

psvo commented:
The patch already adds tags parameter usage example - I modified the original example. Maybe I should add separate example, but I don't feel it necessary.

jsmartin commented:
@psvo Great work. I'm glad someone is adding new features to this module!

psvo committed:
Add Boto version check for tags parameter of cloudformation module

Tags parameter requires at least version 2.6.0 of Boto module. When tags
parameter is used with older version, error is raised. When tags
parameter is unused, module works as before.

psvo committed:
Add version_added to cloudformation tags parameter

psvo commented:
I added boto version check. The version check is clumsy because Boto doen't have good API for it. Check works only for numeric versions (no rc, alpha etc).

While testing changes, I found that anything older than boto 2.10 has problems with current AWS CloudFormation API. Getting  e.g. error msg: time data '2013-09-30T08:43:28.781Z' does not match format '%Y-%m-%dT%H:%M:%SZ' (got while creating/updating VPC).

jctanner commented:
@psvo, looks good, thank you!
